src/sql/workbench/contrib/notebook/find/notebookFindModel.ts

```diff
@@ -585,14 +584,50 @@ export class NotebookFindModel extends Disposable implements INotebookFindModel
 	// In markdown links are defined as [Link Text](https://url/of/the/text). when searching for text we shouldn't
 	// look for the values inside the (), below regex replaces that with just the Link Text.
 	cleanMarkdownLinks(cellSrc: string): string {
-		return cellSrc.replace(/(?:__|[*#])|\[(.*?)\]\(.*?\)/gm, '$1');
+		return cellSrc.replace(/(?:__)|\[([\s\S]*?)\]\(.*?\)/gm, '$1');
 	}
 
 	// remove /n's to calculate the line number to locate the correct element
 	cleanUpCellSource(cellValue: string | string[]): string | string[] {
 		let trimmedCellSrc: string[] = [];
+		const newline = new RegExp(/^([-=]+[\\r\\n]*)$/gm); // new RegExp(/^([\s-=]+[\\r\\n]*)$/gm);
```

Comment:
Get rid of comment.

In general too it would be fantastic to explain what these regex's are supposed to be checking so we can verify that they actually do what they're supposed to.

And for any regex like this we should 100% have unit tests that verify them so we can at least show what cases we've specifically tested/checked. Makes reviewing these a lot easier and safer.

Reply:
agreed! Will add some tests.